src/sentry_core.c

```diff
@@ -1478,8 +1478,11 @@ sentry_clear_attachments(void)
     SENTRY_WITH_SCOPE_MUT (scope) {
         SENTRY_WITH_OPTIONS (options) {
```

Comment:
_Referencing the nested locking assumption in the other comment_: a `SENTRY_WITH_OPTIONS` scope-macro only very shortly acquires a lock, and then only operates on the premise of an incremented ref-count (i.e., options won't be freed) during the scope, whereas `SENTRY_WITH_SCOPE[_MUT]` holds the lock for the entire lexical scope.

Nonetheless, we should change the order here and acquire the options first, and then the scope to have a consistent lock order.